ouroboros-consensus-diffusion/src/ouroboros-consensus-diffusion/Ouroboros/Consensus/NodeKernel.hs

```diff
@@ -426,208 +432,228 @@ forkBlockForging ::
 forkBlockForging IS{..} blockForging =
     forkLinkedWatcher registry threadLabel
     $ knownSlotWatcher btime
-    $ withEarlyExit_ . go
+    $ \currentSlot -> withRegistry (\rr -> withEarlyExit_ $ go rr currentSlot)
```

Comment:
Are we allocating (and closing) a fresh registry every time the current slot changes?

In that case, we don't really need to call `roforkerClose` do we? The registry will handle it soon enough after `go` exits.

Reply:
Broader question: are there any new registries that are long-lived and for which `release` is never called explicitly? That'd be a space leak in the registry itself.

Reply:
> Are we allocating (and closing) a fresh registry every time the current slot changes?
> 
> In that case, we don't really need to call `roforkerClose` do we? The registry will handle it soon enough after `go` exits.

True, it's not strictly necessary to call `roforkerClose`. However, the `ResourceRegistry` documentation mentions: "Typically the registry is only used as a fall-back, ensuring that resources will deallocated even in the presence of exceptions." I've mostly tried to be explicit about releasing resources instead of relying on the registry

Reply:
> Broader question: are there any new registries that are long-lived and for which `release` is never called explicitly? That'd be a space leak in the registry itself.

From what I can recall in the code that I wrote, I've used registries only within limited scopes using `withRegistry`. It would be good to double-check though. And we should also check that we call release on all `ResourceKey`s, which I think you've pointed out is something we forgot to do on at least one occasion: https://github.com/IntersectMBO/ouroboros-consensus/pull/1267#discussion_r1830230665

Reply:
Last time I looked at info-table-profiling, there was no type that was steadily increasing so I would say that we are fine. However, relying on the resource registry closing the tables lead to a surprising behavior. Perhaps it is worth investigating this again, but I would rather defer this investigation until when the node compiles with UTxO-HD. In general both approaches should be denotationally equivalent.
![image](https://github.com/user-attachments/assets/7c07cd09-02fd-4684-9750-9243701fa54f)

Reply:
#1316